src/test/java/org/folio/innreach/domain/listener/KafkaCirculationEventListenerApiTest.java

```diff
@@ -186,6 +188,51 @@ void shouldRenewalLoanToUpdateTransaction() {
 
   }
 
+  @Test
+  @Sql(scripts = {
+    "classpath:db/central-server/pre-populate-central-server.sql",
+    "classpath:db/inn-reach-transaction/pre-populate-inn-reach-transaction.sql",
+  })
+  void shouldUpdateTransactionToCheckIn() {
+    UUID folioLoanId = UUID.fromString("06e820e3-71a0-455e-8c73-3963aea677d4");
+    String date = "2016-07-01T08:20:00.00Z";
+    Instant instant = Instant.parse(date);
+    Date dueDate = Date.from(instant);
+    var event = getLoanDomainEvent(DomainEventType.UPDATED);
+    LoanDTO loanDTO = event.getData().getNewEntity();
+    loanDTO.setId(folioLoanId);
+    loanDTO.setDueDate(dueDate);
+    loanDTO.setAction("checkedin");
+    loanDTO.setStatus(new LoanStatus().name("Closed"));
+    event.setData(new EntityChangedData<>(null, loanDTO));
+
+    event.setRecordId(null); // the listener should set this field from event key value
+
+    var consumerRecord = new ConsumerRecord(CIRC_LOAN_TOPIC, 1, 1, folioLoanId.toString(), event);
+    listener.handleLoanEvents(List.of(consumerRecord));
+
+    ArgumentCaptor<List<DomainEvent<LoanDTO>>> eventsCaptor = ArgumentCaptor.forClass(List.class);
+
+    verify(eventProcessor).process(eventsCaptor.capture(), any(Consumer.class));
+    when(innReachExternalService.postInnReachApi(any(), any(), any())).thenReturn("ok");
+    var capturedEvents = eventsCaptor.getValue();
+    assertEquals(1, capturedEvents.size());
+
+    var capturedEvent = capturedEvents.get(0);
+    assertEquals(folioLoanId, capturedEvent.getRecordId());;
+    Optional<InnReachTransaction> optionalTransaction = transactionRepository.fetchTransactionByLoanId(folioLoanId);
+    InnReachTransaction transaction = null;
+    if (optionalTransaction.isPresent()) {
+      transaction = optionalTransaction.get();
+    }
+    var transactionHold = (TransactionHold) transaction.getHold();
+    var transactionItemHold = (TransactionItemHold) transaction.getHold();
```

Comment:
`var transactionHold = (TransactionItemHold) transaction.getHold();`

Reply:
Got it